### PR TITLE
feat(auth-local/RegisterForm): add 'complexPassword' option

### PR DESCRIPTION
**Pull request by nbhim12**

### CODE REVIEW

File: packages/auth-local/client/components/RegisterForm/utils/joi.js [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
+const defaultEmailValidationSettings = Joi.string()
+  .email({ minDomainSegments: 2, tlds: { allow: false } })
+  .required()
+  .messages({
+    'any.required': 'Fill in the field',
+    'string.empty': 'Fill in the field',
+    'string.email': 'Invalid email'
+  })
+
+const defaulPasswordValidationSettings = Joi.string().required()
```
Comment (cray0000) [inline packages/auth-local/client/components/RegisterForm/utils/joi.js:21]: `default`

please use a spell checker in your IDE. For example for VS Code:
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

**Review by cray0000 — CHANGES_REQUESTED** [review]

File: packages/auth-local/client/components/RegisterForm/utils/joi.js [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
+  confirm: defaultVerifyPasswordValidationSettings
+})
+
+export const complexPasswordSchema = Joi.object().keys({
+  name: defaultNameValidationSettings,
+  email: defaultEmailValidationSettings,
   password: Joi.string().required()
     .min(3)
     .max(32)
+    .regex(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%_*?&#$^])[A-Za-z\d_@$!%*-?&#^]{8,}$/)
```
Comment (cray0000) [inline packages/auth-local/client/components/RegisterForm/utils/joi.js:53]: please add a comment above this line describing what this regex checks for

**Review by cray0000 — CHANGES_REQUESTED** [review]

File: packages/auth-local/client/components/RegisterForm/utils/joi.js (hunk 2 above)
Comment (cray0000) [reply packages/auth-local/client/components/RegisterForm/utils/joi.js:53]: actuallyy nevermind, it is described in the `string. pattern. base` below. I was just confused with the special characters you have in the regex

**Review by cray0000 — CHANGES_REQUESTED** [review]

**Review by cray0000 — APPROVED** [review]

File: packages/auth-local/client/components/RegisterForm/index.js [hunk 3]
```diff
@@ -62,7 +62,8 @@ function RegisterForm ({
   onShouldConfirmRegistration,
   onSuccess,
   onError,
-  onChangeSlide
+  onChangeSlide,
+  complexPassword
```
Comment (zheorekh) [inline packages/auth-local/client/components/RegisterForm/index.js:66]: props must be before callbacks + alphabet order

File: packages/auth-local/client/components/RegisterForm/index.js [hunk 4]
```diff
@@ -211,11 +212,13 @@ function initForm (properties) {
...
 
 RegisterForm.defaultProps = {
-  baseUrl: BASE_URL
+  baseUrl: BASE_URL,
+  passwordCheckType: 'simple'
 }
 
 RegisterForm.propTypes = {
   baseUrl: PropTypes.string,
+  passwordCheckType: PropTypes.string,
```
Comment (zheorekh) [inline packages/auth-local/client/components/RegisterForm/index.js:221]: it doesn't expect any string.

**Review by zheorekh — APPROVED** [review]
LGTM

**Review by cray0000 — APPROVED** [review]

**Pull request merged**